src/constfold.d

```diff
@@ -770,6 +770,10 @@ extern (C++) UnionExp Equal(TOK op, Type type, Expression e1, Expression e2)
             {
                 Expression ee1 = (*es1.elements)[i];
                 Expression ee2 = (*es2.elements)[i];
+                if (!ee1)
+                    ee1 = es1.basis;
```

Comment:
I see this pattern repeated dozens of times in this change. I suggest making a `get` function that retrieves element `i` instead.